pkg/image/builder.go

```diff
@@ -254,29 +243,85 @@ func saveOneImage(srcRegistry registry.RegistryOptions, imagesDir string, image
 		}
 		rewritten, err := rewritePrivateImage(srcRegistry, image, appSlug)
 		if err != nil {
-			return errors.Wrapf(err, "failed to rewrite private image %s", image)
+			return nil, errors.Wrap(err, "failed to rewrite private image")
 		}
 
 		sourceImage = rewritten
 	}
 	srcRef, err := alltransports.ParseImageName(fmt.Sprintf("docker://%s", sourceImage))
 	if err != nil {
-		return errors.Wrap(err, "failed to parse source image name")
+		return nil, errors.Wrapf(err, "failed to parse source image name %s", sourceImage)
+	}
+
+	destCtx := &types.SystemContext{
+		DockerInsecureSkipTLSVerify: types.OptionalBoolTrue,
+	}
+	destCtx.DockerAuthConfig = &types.DockerAuthConfig{
+		Username: destRegistry.Username,
+		Password: destRegistry.Password,
+	}
+
+	destRef, err := alltransports.ParseImageName(fmt.Sprintf("docker://%s", DestRef(destRegistry, image)))
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse dest image name %s", DestRef(destRegistry, image))
 	}
 
 	_, err = copy.Image(context.Background(), policyContext, destRef, srcRef, &copy.Options{
 		RemoveSignatures:      true,
 		SignBy:                "",
 		ReportWriter:          reportWriter,
 		SourceCtx:             sourceCtx,
-		DestinationCtx:        nil,
+		DestinationCtx:        destCtx,
 		ForceManifestMIMEType: "",
 	})
 	if err != nil {
-		return errors.Wrap(err, "failed to copy image")
+		log.Error(errors.Wrap(err, "failed to copy image directly"))
```

Comment:
Suggestion for an improvement here:
If it is specifically that compatibility error, we should log it on info level and retry pull/push approach.
Otherwise we should log this on error level and bail.

Reply:
Would an info level "failed to copy image directly, attempting fallback copy method" be good? I'm not confident in our ability to properly identify that compatibility error, and I don't think optimizing for failure cases is worthwhile in general